### ISSUE TITLE
[Core] Add support for `InCondition` 

### ISSUE THREAD

**Issue by ToshY** [schranz-search/schranz-search#444]
**Suggestion**

The ability to check for multiple values using an `IN` condition would be nice to have. The idea would be roughly be identical to the `EqualCondition` except the second argument allows an array (multiple values).

```php
<?php

use Schranz\Search\SEAL\Search\Condition;

$result = $this->engine->createSearchBuilder()
    ->addIndex('blog')
    ->addFilter(new Condition\InCondition('tags', ['UI','UX','Frontend']))
    ->getResult();
```

---

This probably would require research to check if all engines allow a kind of "IN".

* algolia - [`OR` chaining](https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/in-depth/combining-boolean-operators/#example-scenarios) (unsure)
* elasticsearch - [`terms`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html)
* opensearch - [`terms`](https://opensearch.org/docs/latest/query-dsl/term/terms/)
* loupe - ?
* meilisearch - [`IN`](https://www.meilisearch.com/docs/learn/filtering_and_sorting/filter_expression_reference#in)
* redisearch - [`|` chaining; `@x:(foo|bar)`](https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/query_syntax/#mapping-common-sql-predicates-to-redis-query-engine) (unsure)
* solr - ?
* typesense - ?

**Comment by alexander-schranz** [comment]
Thx for creating the issue.

At the end a `InCondition` is nothing else / a shortcut for a combination of `EqualConditions` inside `OrCondition`. So we could always fallback to that as OrCondition are implemented everywhere if there is no direct support for IN. 

```php
new Condition\OrCondition(
    new Condition\EqualCondition('tags', 'UI'),
    new Condition\EqualCondition('tags', 'UX'),
    new Condition\EqualCondition('tags', 'Frontend'),
);
```

**Comment by alexander-schranz** [comment]
Startet with this in https://github.com/schranz-search/schranz-search/pull/447 and added a reference test.

**Comment by alexander-schranz** [comment]
Implemented in #447
